diff -Naur a/drivers/hwmon/pmbus/ibm-cffps.c b/drivers/hwmon/pmbus/ibm-cffps.c
--- a/drivers/hwmon/pmbus/ibm-cffps.c	2019-09-24 11:38:05.151822715 +0800
+++ b/drivers/hwmon/pmbus/ibm-cffps.c	2019-09-24 11:42:46.897872429 +0800
@@ -363,6 +363,7 @@
 	[cffps1] = {
 		.pages = 1,
 		.func[0] = PMBUS_HAVE_VIN | PMBUS_HAVE_VOUT | PMBUS_HAVE_IOUT |
+			PMBUS_HAVE_IIN | PMBUS_HAVE_POUT |
 			PMBUS_HAVE_PIN | PMBUS_HAVE_FAN12 | PMBUS_HAVE_TEMP |
 			PMBUS_HAVE_TEMP2 | PMBUS_HAVE_TEMP3 |
 			PMBUS_HAVE_STATUS_VOUT | PMBUS_HAVE_STATUS_IOUT |
