diff -Naur a/ipmbbridged.cpp b/ipmbbridged.cpp
--- a/ipmbbridged.cpp	2019-07-21 19:04:31.208175000 +0800
+++ b/ipmbbridged.cpp	2019-07-22 09:38:33.974849086 +0800
@@ -781,6 +781,31 @@
     return returnStatus(ipmbResponseStatus::timeout);
 }
 
+void set_path(std::string &master, std::string &slave)
+{
+        auto MasterHyphen = master.find("-");
+        std::string bus = master.substr(MasterHyphen + 1);
+        int Id = 0;
+        try
+        {
+            Id = std::stoi(bus);
+        }
+        catch (std::invalid_argument)
+        {
+            return;
+        }
+        if (Id == 5)
+        {
+            auto SlaveHyphen = slave.find("-");
+            auto Slavefind = slave.rfind('/', SlaveHyphen);
+            if (SlaveHyphen > Slavefind)
+            {
+                master = master.replace(MasterHyphen + 1, master.length() - MasterHyphen, "11");
+                slave = slave.replace(Slavefind + 1, SlaveHyphen - Slavefind - 1, "11");
+            }
+        }
+}
+
 static IpmbChannel *getChannel(ipmbChannelType channelType)
 {
     auto channel =
@@ -824,6 +849,7 @@
 
             auto channel = ipmbChannels.emplace(ipmbChannels.end(), io, bmcAddr,
                                                 reqAddr, type, commandFilter);
+            set_path((std::string &)masterPath, (std::string &)slavePath);
             if (channel->ipmbChannelInit(slavePath.c_str(),
                                          masterPath.c_str()) < 0)
             {
