diff -Naur a/board/aspeed/ast-g5/ast-g5.c b/board/aspeed/ast-g5/ast-g5.c
--- a/board/aspeed/ast-g5/ast-g5.c	2019-07-31 13:55:04.936121300 +0800
+++ b/board/aspeed/ast-g5/ast-g5.c	2019-08-04 11:13:37.727834403 +0800
@@ -24,10 +24,10 @@
 	bool sdmc_unlocked;
 	u32 val;
 
-	/* iLPC2AHB */
-	val = readl(AST_SCU_BASE + AST_SCU_HW_STRAP1);
+	/* iLPC2AHB  enable decode first*/
+	/*val = readl(AST_SCU_BASE + AST_SCU_HW_STRAP1);
 	val |= SCU_HW_STRAP_LPC_DEC_SUPER_IO;
-	writel(val, AST_SCU_BASE + AST_SCU_HW_STRAP1);
+	writel(val, AST_SCU_BASE + AST_SCU_HW_STRAP1);*/
 
 	val = readl(AST_LPC_BASE + AST_LPC_HICRB);
 	val |= LPC_HICRB_ILPC2AHB;
