diff -Nurd git_ori/presence/tach.cpp git/presence/tach.cpp
--- git_ori/presence/tach.cpp	2019-07-09 17:11:15.267939117 +0800
+++ git/presence/tach.cpp	2019-07-10 09:39:12.892441658 +0800
@@ -84,7 +84,7 @@
             state.end(),
             [](const auto & s)
             {
-                return std::get<int64_t>(s) != 0;
+                return std::get<int64_t>(s) > 0;
             });
 
     return currentState;
@@ -115,7 +115,7 @@
     return std::any_of(
             values.begin(),
             values.end(),
-            [](const auto & v) {return v != 0;});
+            [](const auto & v) {return v > 0;});
 }
 
 void Tach::propertiesChanged(
@@ -147,7 +147,7 @@
                 state.end(),
                 [](const auto & s)
                 {
-                    return std::get<int64_t>(s) != 0;
+                    return std::get<int64_t>(s) > 0;
                 });
 
         if (currentState != newState)
