diff -Naur a/service_files/xyz.openbmc_project.cpusensor.service b/service_files/xyz.openbmc_project.cpusensor.service
--- a/service_files/xyz.openbmc_project.cpusensor.service	2019-10-22 16:45:22.175993970 +0800
+++ b/service_files/xyz.openbmc_project.cpusensor.service	2019-10-22 16:45:55.307754958 +0800
@@ -2,7 +2,7 @@
 Description=CPU Sensor
 StopWhenUnneeded=false
 Requires=xyz.openbmc_project.EntityManager.service
-After=xyz.openbmc_project.EntityManager.service
+After=xyz.openbmc_project.EntityManager.service logrotate.service
 
 [Service]
 Restart=always
