diff -Naur a/writefrudata.cpp b/writefrudata.cpp
--- a/writefrudata.cpp	2019-07-10 10:42:26.262046220 +0800
+++ b/writefrudata.cpp	2019-07-11 21:05:33.646798124 +0800
@@ -211,6 +211,9 @@
         return -1;
     }
 
+    // Workaround for getting wrong service
+    service = "xyz.openbmc_project.Inventory.Manager";
+
     auto iter = frus.find(fruid);
     if (iter == frus.end())
     {
@@ -450,7 +453,7 @@
     // than the offset of any if the FRU areas mentioned in the common header,
     // then we do not have a complete file.
     for (uint8_t fruEntry = IPMI_FRU_INTERNAL_OFFSET;
-         fruEntry < (sizeof(struct common_header) - 2); fruEntry++)
+         fruEntry < (sizeof(struct common_header) - 3); fruEntry++)
     {
         rc = -1;
         // Actual offset in the payload is the offset mentioned in common header
@@ -575,7 +578,7 @@
     FruAreaVector fruAreaVec;
 
     for (uint8_t fruEntry = IPMI_FRU_INTERNAL_OFFSET;
-         fruEntry < (sizeof(struct common_header) - 2); fruEntry++)
+         fruEntry < (sizeof(struct common_header) - 3); fruEntry++)
     {
         // Create an object and push onto a vector.
         std::unique_ptr<IPMIFruArea> fruArea = std::make_unique<IPMIFruArea>(
