diff -Nurd git_ori/sensorhandler.cpp git/sensorhandler.cpp
--- git_ori/sensorhandler.cpp	2019-07-18 19:00:14.079976595 +0800
+++ git/sensorhandler.cpp	2019-07-18 19:08:09.197836603 +0800
@@ -406,21 +406,26 @@
     {
         return IPMI_CC_ILLEGAL_COMMAND;
     }
-
-    try
-    {
-        getResponse = iter->second.getFunc(iter->second);
-        *data_len = getResponse.size();
-        std::memcpy(resp, getResponse.data(), *data_len);
-        resp->operation = 1 << scanningEnabledBit;
-        return IPMI_CC_OK;
-    }
-    catch (const std::exception& e)
+    
+    /* Add retry to wait DBUS response, 
+       without the wait, IPMI might not receive data property if DBUS times out. */
+    for (unsigned int retry = 0; retry < 3; retry++)
     {
-        *data_len = getResponse.size();
-        std::memcpy(resp, getResponse.data(), *data_len);
-        return IPMI_CC_OK;
+        try
+        {
+            getResponse = iter->second.getFunc(iter->second);
+            *data_len = getResponse.size();
+            std::memcpy(resp, getResponse.data(), *data_len);
+            resp->operation = 1 << scanningEnabledBit;
+            break; 
+        }
+        catch (const std::exception& e)
+        {
+            *data_len = getResponse.size();
+            std::memcpy(resp, getResponse.data(), *data_len);
+        }
     }
+    return IPMI_CC_OK;
 }
 
 void getSensorThresholds(uint8_t sensorNum,
