diff -Naur a/read_fru_data.cpp b/read_fru_data.cpp
--- a/read_fru_data.cpp	2019-07-08 18:05:28.656736960 +0800
+++ b/read_fru_data.cpp	2019-07-10 10:23:59.335070197 +0800
@@ -50,6 +50,7 @@
     ipmi::PropertyMap properties;
     sdbusplus::bus::bus bus{ipmid_get_sd_bus_connection()};
     auto service = ipmi::getService(bus, INV_INTF, OBJ_PATH);
+    service = "xyz.openbmc_project.Inventory.Manager"; 
     std::string objPath = OBJ_PATH + path;
     auto method = bus.new_method_call(service.c_str(), objPath.c_str(),
                                       PROP_INTF, "GetAll");
